core/src/main/scala/com/cloudera/spark/CGroupsMemoryHandle.scala

```diff
@@ -0,0 +1,37 @@
+package com.cloudera.spark
+
+import scala.io.Source
+
+class CGroupsMemoryHandle(cgroupMemStats: Seq[(String, Long)]) extends MemoryGetter {
+
+  override val namesAndReporting: Seq[(String, PeakReporting)] = cgroupMemStats.map(x => {
+    (x._1, IncrementBytes)
+  })
+
+  override def values(dest: Array[Long], offset: Int): Unit = {
```

Comment:
when you tried running this, do values of the cgroup metrics actually change at all?

because IIRC, the way this works is that only this method, `values()`, is called repeatedly.  The other stuff is called just once to setup.  If that is right, then you're just returning the same values everytime this method is called.  You need to move the code from your `get()` method into here instead.

Reply:
I don't remember whether they actually changed before my latest commit when this comment was originally made. But in the latest commit the values defineltly changed. I moved the code from get() method to  values() method.